pkg/scm/gitlab/client_actioner.go

```diff
@@ -153,6 +155,81 @@ func (c *Client) ApplyStep(ctx context.Context, evalContext scm.EvalContext, upd
 
 		return err
 
+	case "assign_reviewers":
+		source, err := step.RequiredStringEnum("source", "codeowners")
+		if err != nil {
+			return err
+		}
+
+		desiredLimit, err := step.RequiredInt("limit")
+		if err != nil {
+			return err
+		}
+
+		mode, err := step.OptionalStringEnum("mode", "linear", "linear", "random")
+		if err != nil {
+			return err
+		}
+
+		var eligibleReviewers []scm.Actor
+
+		switch source {
+		case "codeowners":
+			eligibleReviewers = evalContext.GetCodeOwners()
+
+			break
+		}
+
+		if len(eligibleReviewers) == 0 {
+			return nil
+		}
+
+		var reviewers []scm.Actor
+
+		var limit int
+		if desiredLimit > len(eligibleReviewers) {
+			limit = len(eligibleReviewers)
+		} else {
+			limit = desiredLimit
+		}
+
+		switch mode {
+		case "linear":
+			reviewers = eligibleReviewers[:limit]
+
+			break
+		case "random":
+			reviewers = make([]scm.Actor, limit)
+			r := rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
it's fine to use a weak random source here; but I would move it to a package variable so we only initialize the source once in the entire process :)

Reply:
updated